geps/gep-713/index.md

```diff
@@ -1,1157 +1,743 @@
 
-Pro:
+The basic status conditions are:
 
-- This gives object owners a very clear signal that something some Policy is
-  going to affect their object, at apply time, which helps a lot with discoverability.
+* **Accepted**: the meta resource passed both syntactic validation by the API server and semantic validation enforced by the controller, such as whether the target objects exist.
+* **Enforced**: the meta resource’s spec is guaranteed to be fully enforced, to the extent of what the controller can ensure.
+* **Partially enforced**: parts of the meta resource’s spec is guaranteed to be enforced, while other parts are known to have been superseded by other specs, to the extent of what the controller can ensure. The status should include details highlighting which parts of the meta resource are enforced and which parts have been superseded, with the references to all other related meta resources.
+* **Overridden**: the meta resource’s spec is known to have been fully overridden by other specs. The status should include the references to the other related meta resources.
 
-Cons:
+#### Target object status
 
-- Implementations would have to have a webhook, which is another thing to run.
-- The webhook will need to have the same data model that the implementation uses,
-  and keep track of which GatewayClasses, Gateways, Routes, and Policies are
-  relevant. Experience suggests this will not be a trivial engineering exercise,and will add a lot of implementation complexity.
+Implementations of meta resources MUST put a condition into `status.Conditions` of any objects affected by the meta resource.
 
-#### `kubectl` plugin or command-line tool
-To help improve UX and standardization, a kubectl plugin will be developed that
-will be capable of describing the computed sum of policy that applies to a given
-resource, including policies applied to parent resources.
+That condition MUST be named according to the pattern `<meta-resource-kind>Affected` (e.g. `colors.controller.k8s.io/ColorPolicyAffected`), and have the optional `observedGeneration` field kept up to date when the spec of the target object changes.
 
-Each Policy CRD that wants to be supported by this plugin will need to follow
-the API structure defined above and add the [corresponding label](https://gateway-api.sigs.k8s.io/geps/gep-713/#standard-label-on-crd-objects)
-to the CRD.
+Implementations should use their own unique domain prefix for this condition type. Gateway API implementations, for instance, should use the same domain as in the `controllerName` field on `GatewayClass` (or some other implementation-unique domain for implementations that do not use `GatewayClass`.)
 
-### Conditions
+For objects that do not have a `status.Conditions` field available (`Secret` is a good example), that object MUST instead have an annotation of `colors.controller.k8s.io/ColorPolicyAffected: true` added instead.
```

Comment:
We cannot require an implementation to write an annotation to an object they don't own IMO. It may make sense in some cases but I cannot imagine users would be too happy with write access to secrets in most cases for example

Reply:
Can change it to SHOULD.

Reply:
Agreed that SHOULD is better here.